core/src/main/java/org/opensearch/sql/ast/tree/Table.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.ast.tree;
+
+import com.google.common.collect.ImmutableList;
+import java.util.List;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.ToString;
+import org.opensearch.sql.ast.AbstractNodeVisitor;
+import org.opensearch.sql.ast.expression.UnresolvedExpression;
+
+/**
+ * AST node representing a Table operation in the query plan.
+ *
+ * <p>This class represents a table operation in the Abstract Syntax Tree (AST). It contains a list
+ * of expressions to project (select) from the table and can have a child node representing the
+ * source of the data.
+ */
+@ToString
+@Getter
+@EqualsAndHashCode(callSuper = false)
+public class Table extends UnresolvedPlan {
```

Comment:
why not reuse `org.opensearch.sql.ast.tree.Project`? I didn't any difference between `table` and `project` in AST level. I'd like keep as simple as possible in AST.